platform/iphone/in_app_store.mm

```diff
@@ -116,167 +92,150 @@ - (void)productsRequest:(SKProductsRequest *)request didReceiveResponse:(SKProdu
-
-	PoolStringArray pids = params["product_ids"];
-	printf("************ request product info! %i\n", pids.size());
-
-	NSMutableArray *array = [[[NSMutableArray alloc] initWithCapacity:pids.size()] autorelease];
-	for (int i = 0; i < pids.size(); i++) {
-		printf("******** adding %ls to product list\n", pids[i].c_str());
-		NSString *pid = [[[NSString alloc] initWithUTF8String:pids[i].utf8().get_data()] autorelease];
-		[array addObject:pid];
-	};
-
-	NSSet *products = [[[NSSet alloc] initWithArray:array] autorelease];
-	SKProductsRequest *request = [[SKProductsRequest alloc] initWithProductIdentifiers:products];
-
-	ProductsDelegate *delegate = [[ProductsDelegate alloc] init];
-
-	request.delegate = delegate;
-	[request start];
-
-	return OK;
-};
-
-Error InAppStore::restore_purchases() {
-
-	printf("restoring purchases!\n");
-	[[SKPaymentQueue defaultQueue] restoreCompletedTransactions];
-
-	return OK;
-};
-
-@interface TransObserver : NSObject <SKPaymentTransactionObserver> {
-};
-@end
-
-@implementation TransObserver
-
+#pragma mark - SKPaymentTransactionObserver
 - (void)paymentQueue:(SKPaymentQueue *)queue updatedTransactions:(NSArray *)transactions {
-
-	printf("transactions updated!\n");
 	for (SKPaymentTransaction *transaction in transactions) {
+		// Create a variant with this info
+		Dictionary ret = TransactionDictionary(transaction);
+		String pid = String::utf8([transaction.payment.productIdentifier UTF8String]);
 
 		switch (transaction.transactionState) {
 			case SKPaymentTransactionStatePurchased: {
 				printf("status purchased!\n");
-				String pid = String::utf8([transaction.payment.productIdentifier UTF8String]);
-				String transactionId = String::utf8([transaction.transactionIdentifier UTF8String]);
 				InAppStore::get_singleton()->_record_purchase(pid);
-				Dictionary ret;
-				ret["type"] = "purchase";
-				ret["result"] = "ok";
-				ret["product_id"] = pid;
-				ret["transaction_id"] = transactionId;
-
-				NSData *receipt = nil;
-				int sdk_version = 6;
-
-				if ([[[UIDevice currentDevice] systemVersion] floatValue] >= 7.0) {
-
-					NSURL *receiptFileURL = nil;
-					NSBundle *bundle = [NSBundle mainBundle];
-					if ([bundle respondsToSelector:@selector(appStoreReceiptURL)]) {
 
-						// Get the transaction receipt file path location in the app bundle.
-						receiptFileURL = [bundle appStoreReceiptURL];
+				// Get the transaction receipt file path location in the app
+				// bundle and read the contents
+				NSURL *receiptFileURL = [[NSBundle mainBundle] appStoreReceiptURL];
+				NSData *receipt = [NSData dataWithContentsOfURL:receiptFileURL];
 
-						// Read in the contents of the transaction file.
-						receipt = [NSData dataWithContentsOfURL:receiptFileURL];
-						sdk_version = 7;
-
-					} else {
-						// Fall back to deprecated transaction receipt,
-						// which is still available in iOS 7.
-
-						// Use SKPaymentTransaction's transactionReceipt.
-						receipt = transaction.transactionReceipt;
```

Comment:
This is gone in iOS 12.

Reply:
Should fix this current warning (on Travis CI):
```
platform/iphone/in_app_store.mm:209:29: warning: 'transactionReceipt' is deprecated: first deprecated in iOS 7.0 - Use -[NSBundle appStoreReceiptURL] [-Wdeprecated-declarations,29]
                                                receipt = transaction.transactionReceipt;
                                                                       ^
/Applications/Xcode-9.4.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.4.sdk/System/Library/Frameworks/StoreKit.framework/Headers/SKPaymentTransaction.h:47:50: note: 'transactionReceipt' has been explicitly marked deprecated here [2]
 @property(nonatomic, readonly, nullable) NSData *transactionReceipt NS_DEPRECATED_IOS(3_0, 7_0, "Use -[NSBundle appStoreReceiptURL]");
                                                  ^
platform/iphone/in_app_store.mm:213:28: warning: 'transactionReceipt' is deprecated: first deprecated in iOS 7.0 - Use -[NSBundle appStoreReceiptURL] [-Wdeprecated-declarations,29]
                                        receipt = transaction.transactionReceipt;
                                                               ^
/Applications/Xcode-9.4.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.4.sdk/System/Library/Frameworks/StoreKit.framework/Headers/SKPaymentTransaction.h:47:50: note: 'transactionReceipt' has been explicitly marked deprecated here [2]
 @property(nonatomic, readonly, nullable) NSData *transactionReceipt NS_DEPRECATED_IOS(3_0, 7_0, "Use -[NSBundle appStoreReceiptURL]");
                                                  ^
```